hypha/apply/funds/templates/funds/includes/admin_primary_actions.html

```diff
@@ -14,6 +14,12 @@ <h5>Actions to take</h5>
 
 <a data-fancybox data-src="#screen-application" class="button button--bottom-space button--primary button--full-width {% if screening_form.should_show %}is-not-disabled{% else %}is-disabled{% endif %}" href="#">Screen application</a>
 
+{% if object.is_determined %}
```

Comment:
Whats interesting here is that `is_determined` is coming from the submission state and not from the presence of an actual determination. Which is by design, the determination object is only extra data that doesn't affect the state of the object.

I think what needs to happen is that `object.is_determined` is renamed to something else to reflect the state is final or closed and then we want to be confirming that the determination exists as a separate task. So the submission says that its in a state to show determinations as a primary action and then we check the determinations to confirm they exists.

```
{% if object.is_finished %}
    {% with determination=object.determinations.final.last %}
    {% if determination %}
        ...
    {% endif %}
{% endif %}
```

Reply:
Yeah, this makes more sense. The unecessary test changes were the result of my not understanding that an application being finished doesn't necessarily mean it will have a determination. I've made this change and reverted the previously changed tests